File: Data Structures & Algorithms/is-palindrome/submission-0.java
class Solution {
    public boolean isPalindrome(String s) {
        //stack ? put in stack and pop it to see if it the same as every char
        s = s.replaceAll("[^a-zA-Z0-9]", "").toLowerCase();
        int len = s.length();
        if(len == 0){
            return true; 
        }

        Stack<Character> stack = new Stack<>();
        for(char c : s.toCharArray()){
            stack.push(c);
        }

        for(int i=0; i<len;i++){
            if(stack.pop() != s.charAt(i)){
                return false; 
            }
        }

        return true;
    }
}

File: Data Structures & Algorithms/baseball-game/submission-0.py
class Solution:
    def calPoints(self, operations: List[str]) -> int:
        stack = []

        for s in operations:
            if s == '+':
                stack.append(stack[-2] + stack[-1])
            elif s == 'D':
                stack.append(stack[-1] * 2)
            elif s == 'C':
                stack.pop()
            else:
                stack.append(int(s))
        
        res = 0
        while stack:
            res += stack.pop()
        
        return res